test/test_compile.py

```diff
@@ -201,5 +207,17 @@ def test_float8_graph_output(self):
         )
 
 
+@unittest.skipIf(not torch.cuda.is_available() or not is_H100, "CUDA not available")
+def test_sync_amax_func():
+    cnts = CompileCounterWithBackend("inductor")
+    module = torch.nn.Sequential(
+        nn.Linear(16, 32, bias=True), nn.ReLU(), nn.Linear(32, 16, bias=True)
+    )
+    float8_mod = swap_linear_with_float8_linear(module, Float8DynamicLinear)
+    compiled_swap_func = torch.compile(sync_float8_amax_and_scale_history, backend=cnts)
+    compiled_swap_func(float8_mod)
+    assert cnts.frame_count == 0, "Compiled graph should have 1 frame!"
```

Comment:
This is not intuitive to me. I was curious why the frame count is not 1?

Reply:
This is not intuitive to me either and actually this test is not really doing what I thought it was doing, I talked to Brian yesterday. What I think I think is happening is that the outer compile which I am checking sees the inner compiled region and just backtracks out not compiling anything. So this test doesn't really do anything..